src/Kestrel.Core/Internal/FrameConnection.cs

```diff
@@ -10,15 +10,15 @@
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Hosting.Server;
 using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Protocols.Features;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http;
 using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
-using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.Internal;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
```

Comment:
The other thing is that very little of this file has anything to do with HTTP. Everything to do with connection/socket lifetime, connection adapters, scopes, logging, efficient timeout management, etc. will likely be abstracted out to the connection level. But not yet I guess... @davidfowl what are you waiting for? 😝